src/main/scala/com/microsoft/hyperspace/index/rules/RuleUtils.scala

```diff
@@ -76,29 +73,28 @@ object RuleUtils {
       }
     }
 
-    def isHybridScanCandidate(
-        entry: IndexLogEntry,
-        inputSourceFiles: Seq[FileInfo],
-        inputSourceFilesSizeInBytes: Long): Boolean = {
-      // TODO: Some threshold about the similarity of source data files - number of common files or
-      //  total size of common files.
-      //  See https://github.com/microsoft/hyperspace/issues/159
+    val hybridScanEnabled = HyperspaceConf.hybridScanEnabled(spark)
+    val hybridScanDeleteEnabled = HyperspaceConf.hybridScanDeleteEnabled(spark)
+
+    def getHybridScanCandidate(index: IndexLogEntry): Option[IndexLogEntry] = {
       // TODO: As in [[PlanSignatureProvider]], Source plan signature comparison is required to
       //  support arbitrary source plans at index creation.
       //  See https://github.com/microsoft/hyperspace/issues/158
 
-      entry.withCachedTag(relation.plan, IndexLogEntryTags.IS_HYBRIDSCAN_CANDIDATE) {
+      val entry = relation.closestIndexVersion(index)
```

Comment:
`cloestIndexVersion` sounds a bit weird since it returns an entry. Maybe just `closestIndex`?